### ISSUE TITLE
prefer-destructuring - false positive on property assignment (with "enforceForRenamedProperties")

### ISSUE THREAD

**Issue by OliverSieweke** [eslint/eslint#11836]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 10.16.0 (current LTS)
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

Minimal working example for `.eslintrc.json`
```js
{
    "parserOptions": {
        "ecmaVersion": 2018
    },
    "rules": {
        "prefer-destructuring": [
            "error",
            {
                "array": true,
                "object": true
            },
            {
                "enforceForRenamedProperties": true
            }
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

`test.js`:
```js
let obj1, obj2;

obj1.property1 = obj2.property2;
```

```bash
eslint test.js
```

**What did you expect to happen?** 

No error, as it is not possible to destructure a property assignment.

**What actually happened? Please include the actual, raw output from ESLint.**
 ``` 
3:1  error  Use object destructuring  prefer-destructuring

✖ 1 problem (1 error, 0 warnings)
```

**Comments**

The problem only arises when the `enforceForRenamedProperties` option is set to `true`.

I'm happy to submit pull request if this is recognized as a bug.

**Comment by kaicataldo** [comment]
Reopening so that we can verify if this is a bug or not

**Comment by mdjermanovic** [comment]
This works as intended since the code can be rewritten to use object destructuring:

```js
let obj1, obj2;

({ property2: obj1.property1 } = obj2);
```
